x-pack/plugins/maps/public/connected_components/edit_layer_panel/edit_layer_panel.tsx

```diff
@@ -244,12 +245,12 @@ export class EditLayerPanel extends Component<Props, State> {
     }
 
     const descriptor = this.props.selectedLayer.getDescriptor() as VectorLayerDescriptor;
-    const numberOfJoins = descriptor.joins ? descriptor.joins.length : 0;
     return isLayerGroup(this.props.selectedLayer)
       ? null
       : this.props.selectedLayer.renderSourceSettingsEditor({
           currentLayerType: this.props.selectedLayer.getType(),
-          numberOfJoins,
+          hasSpatialJoins: (descriptor.joins ?? []).some(isSpatialJoin),
```

Comment:
unmounting happens because `key` is to track selected layer. This occurs here https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/connected_components/edit_layer_panel/index.ts#L23

This should be the case with main and nothing really changed, just the method name.

Reply:
I looked into this more, by logging console statement in EditLayerPanel.componentDidMount and verified that changing scaling type does not trigger another mount.

I think the scroll bar change you are seeing is the change in height of the scaling panel. Notice how `Limit` shows an additional switch, this causes the panel size to change
<img width="200" alt="Screen Shot 2023-05-05 at 8 17 15 AM" src="https://user-images.githubusercontent.com/373691/236483847-68e32cfe-348c-4221-96cd-465dfac4d601.png">
<img width="200" alt="Screen Shot 2023-05-05 at 8 17 21 AM" src="https://user-images.githubusercontent.com/373691/236483851-2705eea0-5f13-41af-82b6-dd8ecd749ea6.png">